auth/src/main/kotlin/com/hedvig/android/auth/network/SubmitAuthorizationCodeResponse.kt

```diff
@@ -0,0 +1,43 @@
+package com.hedvig.android.auth.network
+
+import com.hedvig.android.auth.AccessToken
+import com.hedvig.android.auth.AuthTokenResult
+import com.hedvig.android.auth.RefreshToken
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Response
+
+@Serializable
+data class SubmitAuthorizationCodeResponse(
+  @SerialName("access_token")
+  val accessToken: String,
+  @SerialName("expires_in")
+  val accessTokenExpiresIn: Int,
+  @SerialName("refresh_token")
+  val refreshToken: String,
+  @SerialName("refresh_token_expires_in")
+  val refreshTokenExpiresIn: Int,
+)
+
+fun Response.toAuthTokenResult(): AuthTokenResult {
+  val responseBody = body?.string()
+  val result = if (isSuccessful && responseBody != null) {
+    Json.decodeFromString<SubmitAuthorizationCodeResponse>(responseBody).toAuthAttemptResult()
+  } else {
+    AuthTokenResult.Error(message)
+  }
+  return result
+}
+
+private fun SubmitAuthorizationCodeResponse.toAuthAttemptResult() = AuthTokenResult.Success(
+  accessToken = AccessToken(
+    token = accessToken,
+    expiryInSeconds = accessTokenExpiresIn,
+  ),
+  refreshToken = RefreshToken(
+    token = refreshToken,
+    expiryInSeconds = refreshTokenExpiresIn,
```

Comment:
A thought again. I'm not sure exactly how we'll be using these expiration times, but I'd imagine that having the time in a number won't be that useful by itself, since when checking it out later it won't mean much if we don't also know the starting time of when this will expire.
I wonder if it's a good idea to take the milliseconds it will expire in, and at this level immediately truth it into a kotlinx.datetime.Instant object (or kotlinx.datetime.WhateverSuitsThis) so that later we can always compare that with current time 🤔

Reply:
Good idea! Ill add that in another PR when we actually use these if thats ok :)

Reply:
Yeah and *if* we use them too, we might not if we don't want to do anything special with them and simply try with whatever we have, and re-fetch if it's gonna expire.